src/transformers/integrations/tensor_parallel.py

```diff
@@ -878,11 +878,22 @@ def _prepare_output_fn(output_layouts, use_local_output, mod, outputs, device_me
         Here we are just making each rank believe that he is alone, and he computes his part of the hiddenstates.
         """
```

Comment:
can we just add something here that explains that we need a masking class, and why!